addons/edi_notifier/models/email_notifier.py

```diff
@@ -17,8 +17,12 @@ def _should_notify(self, notifier, event_type, rec):
 
     @api.multi
     def _notify(self, notifier, event_type, recs):
+        instance_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        context = self.env.context.copy()
+        context.update({'instance_url': instance_url})
         for rec in recs:
-            template = notifier.template_id
+            
+            template = notifier.template_id.with_context(context)
```

Comment:
Probably you don't need the copy of the context of line 21, instead you can do:
```
template = notifier.template_id.with_context(instance_url=instance_url)
```
Which should create a new context.